src/passes/identifierManglerPass/declarationNameMangler.ts

```diff
@@ -55,28 +55,43 @@ export const reservedTerms = [
   'const',
   'struct',
   'namespace',
-];
+]);
 
 const unsupportedCharacters = ['$'];
 
 export function checkSourceTerms(term: string, node: ASTNode) {
-  if (reservedTerms.includes(term)) {
+  if (reservedTerms.has(term)) {
     throw new WillNotSupportError(`${printNode(node)} contains ${term} which is a cairo keyword`);
   }
 
-  unsupportedCharacters.forEach((c: string) => {
-    if (term.includes(c)) {
-      throw new WillNotSupportError(
-        `${printNode(node)} ${term} contains unsupported character ${c}`,
-      );
-    }
-  });
+  // Creating the regular expression that match unsupportedCharacters
+  let regex_str = '\\' + unsupportedCharacters[0];
+  for (let index = 1; index < unsupportedCharacters.length; index++) {
+    regex_str += '|\\' + unsupportedCharacters[index];
+  }
+  // Looking for possible matches
+  const regex = RegExp(regex_str, 'g');
+  let match;
+  let unsupported_characters_found = '';
+  while ((match = regex.exec(term)) !== null) {
```

Comment:
Does `term` should get updated inside the body of the loop?

Reply:
The goal is to find all occurrence of unsupported characters in that term, the flag 'g' when created the regex do that. When no other match is found the regex.exec(term) returns null. Term is the string where the look is made. Why would you need it to change?